Tools/TestWebKitAPI/Tests/WebKitCocoa/RestoreLocalStorage.mm

```diff
@@ -50,32 +61,73 @@ static void testFetchLocalStorage(RetainPtr<WKWebsiteDataStore> websiteDataStore
     [webView loadHTMLString:@"<body>webkit.org</body>" baseURL:[NSURL URLWithString:@"https://webkit.org"]];
     [navigationDelegate waitForDidFinishNavigation];
 
-    __block bool doneEvaluatingJavaScript = false;
+    // Put data into local storage.
     [webView evaluateJavaScript:@"localStorage.setItem('key', 'value');" completionHandler:^(id value, NSError *error) {
         EXPECT_NULL(error);
-        doneEvaluatingJavaScript = true;
+        done = true;
     }];
-    TestWebKitAPI::Util::run(&doneEvaluatingJavaScript);
+    TestWebKitAPI::Util::run(&done);
+    done = false;
 
+    // Fetch the local storage data.
     RetainPtr set = [NSSet setWithObject:WKWebsiteDataTypeLocalStorage];
-
-    __block bool doneFetching = false;
+    __block RetainPtr<NSData> localStorageData;
     [[[webView configuration] websiteDataStore] _fetchDataOfTypes:set.get() completionHandler:^(NSData *data) {
         EXPECT_NOT_NULL(data);
         EXPECT_GT([data length], 0u);
-        doneFetching = true;
+        localStorageData = data;
+        done = true;
     }];
-    TestWebKitAPI::Util::run(&doneFetching);
+    TestWebKitAPI::Util::run(&done);
+    done = false;
+
+    // Restore Local Storage.
+
+    // Clear the data store.
+    [websiteDataStore removeDataOfTypes:[WKWebsiteDataStore allWebsiteDataTypes] modifiedSince:[NSDate distantPast] completionHandler:^{
+        done = true;
+    }];
+    TestWebKitAPI::Util::run(&done);
+    done = false;
+
+    // Create a new WebView.
+    RetainPtr newViewConfiguration = adoptNS([WKWebViewConfiguration new]);
```

Comment:
You don't need to create new view configuration or new navigation delegate. You could reuse the existing ones for the new web view.

Reply:
Ah you're right, that makes the test a bit simpler. Thank you!